# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546401126 28800
#      Tue Jan 01 19:52:06 2019 -0800
# Branch stable
# Node ID d9e7affb27c821afc9e6bae49f8218588d717a81
# Parent  09a793a92c11dc8a66d0e33bca139026fdc7d23f
rand.cc: Fix static analyzer detected issues (bug #55347).

* rand.cc (do_rand): Don't use tmp.double_value () twice when intermediate
variable dval already contains this quantity.  Don't call nint_big twice for
the same number.

diff --git a/libinterp/corefcn/rand.cc b/libinterp/corefcn/rand.cc
--- a/libinterp/corefcn/rand.cc
+++ b/libinterp/corefcn/rand.cc
@@ -172,18 +172,17 @@ do_rand (const octave_value_list& args, 
           {
             double dval = tmp.double_value ();
 
             if (octave::math::isnan (dval))
               error ("%s: NaN is invalid matrix dimension", fcn);
 
             dims.resize (2);
 
-            dims(0) = octave::math::nint_big (tmp.double_value ());
-            dims(1) = octave::math::nint_big (tmp.double_value ());
+            dims(0) = dims(1) = octave::math::nint_big (dval);
 
             goto gen_matrix;
           }
         else if (tmp.is_range ())
           {
             Range r = tmp.range_value ();
 
             if (! r.all_elements_are_ints ())
